ra2ce/graph/graph_files/graph_files_collection.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from pathlib import Path
+
+from geopandas import GeoDataFrame
+from networkx import MultiGraph
+
+from ra2ce.graph.graph_files.graph_file import GraphFile
+from ra2ce.graph.graph_files.graph_files_protocol import GraphFileProtocol
+from ra2ce.graph.graph_files.network_file import NetworkFile
+
+
+@dataclass
+class GraphFilesCollection:
+    base_graph: GraphFile = field(
+        default_factory=lambda: GraphFile(name="base_graph.p")
+    )
+    base_graph_hazard: GraphFile = field(
+        default_factory=lambda: GraphFile(name="base_graph_hazard.p")
+    )
+    origins_destinations_graph: GraphFile = field(
+        default_factory=lambda: GraphFile(name="origins_destinations_graph.p")
+    )
+    origins_destinations_graph_hazard: GraphFile = field(
+        default_factory=lambda: GraphFile(name="origins_destinations_graph_hazard.p")
+    )
+    base_network: NetworkFile = field(
+        default_factory=lambda: NetworkFile(name="base_network.feather")
+    )
+    base_network_hazard: NetworkFile = field(
+        default_factory=lambda: NetworkFile(name="base_network_hazard.feather")
+    )
+
+    @property
+    def _graph_collection(self) -> list[GraphFile | NetworkFile]:
+        return [
+            self.base_graph,
+            self.base_graph_hazard,
+            self.origins_destinations_graph,
+            self.origins_destinations_graph_hazard,
+            self.base_network,
+            self.base_network_hazard,
+        ]
+
+    def has_graphs(self) -> bool:
+        """
+        Tests if any of the types has a graph already read
+
+        Returns:
+            bool: True if any graph already read
+        """
+        return any(
+            iter([x for x in self._graph_collection if x.graph is not None]),
+        )
+
+    def get_graph_file(self, graph_file_type: str) -> GraphFileProtocol:
```

Comment:
This is only being used internally, you might as well rename it to 
```suggestion
    def _get_graph_file(self, graph_file_type: str) -> GraphFileProtocol:
```

Reply:
Agree